### ISSUE TITLE
Custom PDF export

### ISSUE THREAD

**Issue by extralucide** [typora/typora-issues#2948]
Hello
I write my doc with `Typora` and I think it's a **great** tool.
I also use **pandoc** alone with my own **latex** template and to concatenate multiple `.md` files in one PDF document with my own header and footer.

Could you add 4 more feature in a future version:

- to export a PDF with our own latex template ?
- to create a PDF from multiples sources ?
- Use a custom **pandoc** command to add filter (for graphviz for instance)?
- Interpret Graphviz in code block ?

Thx

**Comment by aurelien-opex** [comment]
Indeed, I support the idea.

Currently, when I write long documents based on multiple markdown files I have to write a specific script for configuring how Pandoc will assemble them to make a pdf document.
It would be nice if we could edit such script within Typora, then call Pandoc from Typora and launch a PDF viewer from Typora as well. 

That way, Typora could continue to focus on markdown editing. The above feature would just an extension to control the workflow, without adding tricky features like joining multiple document, which is already handled smoothly by Pandoc.

**Comment by TheFern2** [comment]
@extralucide @aurelien-opex  my vscode extension does everything you've asked for. I wrote this specific extension because I am writing a book, but so far it looks like people are liking it too. Give it a try and if you have any issues/requests plz open an issue or shoot me an email fernanbobe+git@protonmail.com

https://marketplace.visualstudio.com/items?itemName=TheFern.vscode-gutenberg

- to export a PDF with our own latex template ?
    - Yes, give your template parameter in the config pandocCommandExtra
- to create a PDF from multiples sources ?
    - Yes, has a gui interface to select files, or commands based on structure
- Use a custom pandoc command to add filter (for graphviz for instance)?
    - Yes, give your filter parameter in the config pandocCommandExtra, I've tested with mermaid previously
- Interpret Graphviz in code block ?
    - I don't personally use graphviz but vscode has preview extensions

Basically if your machine can run the pandoc template and filter, then you can feed the same to the extension.

**Comment by abnerlee** [comment]
https://support.typora.io/Export/#pdf-latex--pandoc
